### PR TITLE
Add MLST wrapper

### PR DESCRIPTION
**Pull request by maxlcummins**
### Description

Add a wrapper script for MLST

### QC

For all wrappers added by this PR, I made sure that

* [x] there is a test case which covers any introduced changes,
* [x] `input:` and `output:` file paths in the resulting rule can be changed arbitrarily,
* [x] all `environment.yaml` specifications follow [the respective best practices](https://stackoverflow.com/a/64594513/2352071),
* [x] wherever possible, command line arguments are inferred and set automatically (e.g. based on file extensions in `input:` or `output:`),
* [x] all fields of the example rules in the `Snakefile`s and their entries are explained via comments (`input:`/`output:`/`params:` etc.),
* [x] `stderr` and/or `stdout` are logged correctly (`log:`), depending on the wrapped tool,
* [x] the `meta.yaml` contains a link to the documentation of the respective tool or command,
* [x] `Snakefile`s pass the linting (`snakemake --lint`),
* [x] `Snakefile`s are formatted with [snakefmt](https://github.com/snakemake/snakefmt),
* [x] Python wrapper scripts are formatted with [black](https://black.readthedocs.io).

### CODE REVIEW

**Comment by maxlcummins** [comment]
Hello, having looked at the log I am unsure of what is causing the check above to fail. Any ideas of how I can resolve this issue please?

**Comment by maxlcummins** [comment]
Hi there @jafors, is there any chance you can please give me a hand with this one when you are free?

**Review by jafors — CHANGES_REQUESTED** [review]
Hey, this tests are passing now, you just ran into some connection problems by github, I guess.
Try to clean the wrapper following the points I made on your last entry, especially reduce the `environment.yaml`. ;)

**Comment by maxlcummins** [comment]
I typically export my environments from conda using `conda env export` which adds in all the extras - I didn't realise you could provide such a succinct yaml until recently thanks to your reviews. Thanks!

File: bio/mlst/test/Snakefile [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        #   --datadir [X]     PubMLST data
+        # HOMEPAGE
+        #   https://github.com/tseemann/mlst - Torsten Seemann
+        extra="--nopath",
+    log:
+        "logs/{sample}.mlst.log",
+    threads: 1
+    wrapper:
+        "0.2.0/bio/mlst"
+        #"https://raw.githubusercontent.com/maxlcummins/snakemake-wrappers/mlst/bio/mlst/wrapper.py"
```
Comment (jafors) [inline bio/mlst/test/Snakefile:50]:
```suggestion
        "master/bio/mlst"
```

**Review by jafors — CHANGES_REQUESTED** [review]
Sure, no problem. Some suggestions similar to the last wrapper.

**Comment by maxlcummins** [comment]
Cheers, the next wrapper I submit will have log files and environment files changed as per your requests :)

File: bio/mlst/wrapper.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+from os import path
+
+log = snakemake.log_fmt_shell(stdout=False, stderr=True)
+
+shell(
+    "mlst"
+    " {snakemake.params.extra}"
+    " {snakemake.input.assembly}"
+    " > {snakemake.output.mlst}"
+    "{log}"
```
Comment (jafors) [inline bio/mlst/wrapper.py:16]:
```suggestion
    " {log}"
```
I forgot the space in my suggestion ;)

**Review by jafors — APPROVED** [review]

**Pull request merged**